brigade-generic-gateway/cmd/brigade-generic-gateway/server_test.go

```diff
@@ -55,3 +55,49 @@ func TestNewRouter(t *testing.T) {
 		t.Fatalf("Expected 401 status, got: %s", res.Status)
 	}
 }
+
+func TestNewRouterCloudEvent(t *testing.T) {
```

Comment:
Regarding a single set of test cases for both event types, I'm not sure whether this would be ideal, since CloudEvents has required properties etc. and we might want to do special checks on them (e.g. specversion which must be '0.2'). 
However, many thanks for the `iterates over an array of structs` proposal, just implemented that!
Only one caveat: if I do `t.Parallel()` in [this](https://github.com/dgkanatsios/brigade/blob/cloudevents/pkg/webhook/genericsimpleevent_test.go#L163) line, I had weird results :( Need to check it out, but I suppose it does not block this PR